src/main/webapp/app/components/Tabs/CurationHistoryTab.tsx

```diff
@@ -0,0 +1,259 @@
+        }
+      }
+      return options;
+    }, [usersData]);
+
+    useEffect(() => {
+      const cleanupCallbacks = [];
+      cleanupCallbacks.push(addUsersListener(firebaseUsersPath));
+      cleanupCallbacks.push(addDrugListListener());
+
+      return () => {
+        cleanupCallbacks.forEach(callback => callback && callback());
+
+        historyTabStore.reset();
+      };
+    }, []);
+
+    function getHistoryContent(historyTabData: HistoryTabData[], maxLength: number = null) {
+      const filteredData = historyTabData.filter(data => {
+        const author = historyTabStore.appliedAuthor;
+        const startDate = historyTabStore.appliedStartDate && new Date(historyTabStore.appliedStartDate);
+        const endDate = historyTabStore.appliedEndDate && new Date(historyTabStore.appliedEndDate);
+
+        let matchesAuthor = false;
+        let matchesDates = false;
+        const timeStamp = new Date(data.timeStamp);
+        if (!author || author.label === data.admin || author.label === data.record.lastEditBy) {
+          matchesAuthor = true;
+        }
+        if (startDate && endDate) {
+          matchesDates = timeStamp >= startDate && timeStamp <= endDate;
+        } else if (startDate) {
+          matchesDates = timeStamp >= startDate;
+        } else if (endDate) {
+          matchesDates = timeStamp <= endDate;
+        } else {
+          matchesDates = true;
+        }
+        return matchesAuthor && matchesDates;
+      });
+
+      // CONSTRUCT TIME SERIES DATA
+      const eventData: (RequiredTimeSeriesEventData | ExtraTimeSeriesEventData)[] = [];
+      for (const data of filteredData) {
+        const timeSeriesData = constructTimeSeriesData(data.record, data.admin, data.timeStamp, data.objectField);
+        if (timeSeriesData) {
+          eventData.push(timeSeriesData);
+        }
+
+        if (maxLength && eventData.length >= maxLength) {
+          break;
+        }
+      }
+
+      // CONSTRUCT HISTORY TAB CONTENT
+      const content: JSX.Element[] = [];
+      for (const data of eventData) {
+        const isContentAvailable = data.content['props']?.children ? true : false;
+
+        content.push(
+          <div>
+            <Row className="justify-content-between" style={{ marginBottom: '22px' }}>
+              <span>
+                <TextFormat value={data.createdAt} type="date" format={APP_HISTORY_FORMAT} />
+                <span className="mr-2" />
+                <TextFormat value={data.createdAt} type="date" format={APP_TIME_FORMAT} />
+              </span>
+              <DefaultTooltip
+                overlayStyle={{ maxWidth: 600 }}
+                overlayInnerStyle={{ maxHeight: '400px', overflow: 'auto', maxWidth: 'inherit' }}
+                overlay={isContentAvailable ? data.content : 'No content available'}
+              >
+                <FaHistory className={!isContentAvailable && 'grey-tooltip'} />
+              </DefaultTooltip>
```

Comment:
In TimeSeries, you have a default tooltip when history is not available. Should we consolidate?

Reply:
Since I'm not using the TimeSeries component in the sidebar (since I could not remove the line on the left side) or the GeneHistoryTooltip, the only overlap between the two types of history we have `GeneHistoryTooltip` and `CurationHistoryTab` are their use of the utility functions. To do this, I would need to refactor all of this, so I think it is best to leave for now and refactor in another PR.